gateleen-hook/src/test/java/org/swisspush/gateleen/hook/HookHandlerTest.java

```diff
@@ -641,6 +644,144 @@ public void hookRegistration_RouteWithRouteMultiplier(TestContext testContext) t
+
+        // Mock the route and listener
+        Route mockRoute = Mockito.mock(Route.class);
+        HttpHook mockHook = new HttpHook("destination/matching");
+        Mockito.when(mockRoute.getHook()).thenReturn(mockHook);
+
+        Listener mockListener = new Listener("listener1", "monitoredUrl", "destination/matching", mockHook);
+
+        // Mock repositories
+        ListenerRepository listenerRepository = Mockito.mock(ListenerRepository.class);
+        RouteRepository routeRepository = Mockito.mock(RouteRepository.class);
+
+        // Configure mocked behavior for repositories with no matching routes or listeners
+        Mockito.when(routeRepository.getRoutes()).thenReturn(Map.of());
+        Mockito.when(listenerRepository.getListeners()).thenReturn(Collections.emptyList());
+
+        // Create HookHandler instance
+        HookHandler hookHandler = new HookHandler(vertx, httpClient, storage, loggingResourceManager,
+                logAppenderRepository, monitoringHandler, "userProfilePath", HOOK_ROOT_URI, requestQueue,
+                false
+        );
+
+        // Use reflection to set private fields
+        setPrivateField(hookHandler, "listenerRepository", listenerRepository);
+        setPrivateField(hookHandler, "routeRepository", routeRepository);
+
+        // Call the method under test
+        hookHandler.handleHookSearch("destination", response);
+
+        // Capture the output and verify the response was sent correctly
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        Mockito.verify(response).end(captor.capture());
+
+        // Check the result
+        String capturedResult = captor.getValue();
+        JsonObject result = new JsonObject(capturedResult);
+        JsonArray routes = result.getJsonArray("routes");
+        JsonArray listeners = result.getJsonArray("listeners");
+
+        // Assert that there are no matching routes or listeners
+        context.assertTrue(routes.isEmpty());
+        context.assertTrue(listeners.isEmpty());
+
+        // Verify the content-type header was set correctly
+        Mockito.verify(response).putHeader("content-type", "application/json");
+    }
+
+    @Test
+    public void testHookHandleSearchWithInvalidQueryParam(TestContext context) {
+        // Mocking the HttpServerResponse and request objects
+        HttpServerResponse response = Mockito.mock(HttpServerResponse.class);
+        Mockito.when(response.putHeader(anyString(), anyString())).thenReturn(response);
+
+        HttpServerRequest request = Mockito.mock(HttpServerRequest.class);
+        Mockito.when(request.response()).thenReturn(response);
+        Mockito.when(request.getParam("q")).thenReturn(null);
+
+        // Call hookHandleSearch
+        hookHandler.handleHookSearch(null, response);
+
+        // Verify that nothing is returned
+        Mockito.verify(response, Mockito.never()).end(any(Buffer.class));
```

Comment:
What happens to the original request when `response.end()` is never called?

Reply:
I will recreate this test, and cover it.

Reply:
We have a timeout in place, but I'll be refining this approach.